### PR TITLE
feat(metrics): deployment counters

### PR DESCRIPTION
**Pull request by ig-ra**

### CODE REVIEW

File: internal/backend/deployments/deployments.go [hunk 1]
```diff
@@ -144,6 +154,27 @@ func drain(ctx context.Context, tx db.DB, id sdktypes.DeploymentID) error {
...
+		case sdktypes.DeploymentStateActive:
+			deploymentsActiveCounter.Add(ctx, val)
+		case sdktypes.DeploymentStateDraining:
+			deploymentsDrainingCounter.Add(ctx, val)
+		case sdktypes.DeploymentStateInactive:
+			deploymentsInactiveCounter.Add(ctx, val)
+		}
+	}
+	updateStateCounter(state, 1)
+	updateStateCounter(oldState, -1)
```
Comment (efiShtain) [inline internal/backend/deployments/deployments.go:173]: you can't update a counter with -1
maybe you can technically
but a counter should represent and ever increasing value

a gauge or an up/down counter should represent a varying value

Comment (ig-ra) [reply internal/backend/deployments/deployments.go:173]: it's an up/down counter

Comment (ig-ra) [reply internal/backend/deployments/deployments.go:173]: do you want me to change name to represent it?

Comment (itayd) [reply internal/backend/deployments/deployments.go:173]: i guess otel supports that?

File: internal/backend/deployments/metrics.go [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+var (
+	deploymentsActiveCounter   metric.Int64UpDownCounter
+	deploymentsDrainingCounter metric.Int64UpDownCounter
+	deploymentsCreatedCounter  metric.Int64Counter
+)
+
+func initMetrics(t *telemetry.Telemetry) {
+	deploymentsActiveCounter, _ = t.NewUpDownCounter("deployments.activated", "Activated deployments counter")
+	deploymentsDrainingCounter, _ = t.NewUpDownCounter("deployments.drained", "Drained deployments counter")
```
Comment (efiShtain) [inline internal/backend/deployments/metrics.go:18]: draining

File: internal/backend/deployments/metrics.go [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+// deployment lifecycle: inactive (upon creation) -> active (upon activation) -> draining (optional) -> inactive
+
+var (
+	deploymentsActiveCounter   metric.Int64UpDownCounter
+	deploymentsDrainingCounter metric.Int64UpDownCounter
+	deploymentsCreatedCounter  metric.Int64Counter
+)
+
+func initMetrics(t *telemetry.Telemetry) {
+	deploymentsActiveCounter, _ = t.NewUpDownCounter("deployments.activated", "Activated deployments counter")
```
Comment (efiShtain) [inline internal/backend/deployments/metrics.go:17]: please change to active

File: internal/backend/deployments/metrics.go (hunk 2 above)
Comment (efiShtain) [reply internal/backend/deployments/metrics.go:18]: rename counter to gauge when it's an up down counter

**Review by efiShtain — APPROVED** [review]
merge after fix naming

**Pull request merged**